src/IB/IBFEInstrumentPanel.cpp

```diff
@@ -927,161 +459,314 @@ IBFEInstrumentPanel::getNumberOfMeterMeshes() const
+            }
         }
+    }
 
-        // set dofs in meter mesh to correspond to the same values
-        // as in the parent mesh
-        for (unsigned int d = 0; d < NDIM; ++d)
+    // Broadcast the local data.
+    IBTK_MPI::sumReduction(x_meter_dofs.data(), n_dofs);
+    IBTK_MPI::sumReduction(u_meter_dofs.data(), n_dofs);
+
+    // Compute the mean values and store them in the vector of DOFs.
+    std::vector<double> x_mean(NDIM, 0.0), u_mean(NDIM, 0.0);
+    for (unsigned int i = 0; i < d_num_perim_nodes[meter_idx]; ++i)
+    {
+        for (int d = 0; d < NDIM; ++d)
         {
-            const int vel_dof_idx = node->dof_number(velocity_sys_num, d, 0);
-            velocity_coords.set(vel_dof_idx, U_dofs[d]);
-            const int disp_dof_idx = node->dof_number(displacement_sys_num, d, 0);
-            displacement_coords.set(disp_dof_idx, dX_dofs[d]);
+            x_mean[d] += x_meter_dofs[NDIM * i + d] / static_cast<double>(d_num_perim_nodes[meter_idx]);
+            u_mean[d] += u_meter_dofs[NDIM * i + d] / static_cast<double>(d_num_perim_nodes[meter_idx]);
         }
     }
-
-    // set dofs for the centroid node in the meter mesh
-    const Node* centroid_node = &d_meter_meshes[meter_mesh_number]->node_ref(d_num_nodes[meter_mesh_number]);
-    for (unsigned int d = 0; d < NDIM; ++d)
+    for (int d = 0; d < NDIM; ++d)
     {
-        const int vel_dof_idx = centroid_node->dof_number(velocity_sys_num, d, 0);
-        velocity_coords.set(vel_dof_idx, mean_U_dofs[d]);
-        const int disp_dof_idx = centroid_node->dof_number(displacement_sys_num, d, 0);
-        displacement_coords.set(disp_dof_idx, mean_dX_dofs[d]);
+        x_meter_dofs[NDIM * d_num_perim_nodes[meter_idx] + d] = x_mean[d];
+        u_meter_dofs[NDIM * d_num_perim_nodes[meter_idx] + d] = u_mean[d];
     }
 
-    // also populate solution vector in the system
-    MeshBase::const_node_iterator node_it = d_meter_meshes[meter_mesh_number]->local_nodes_begin();
-    const MeshBase::const_node_iterator end_node_it = d_meter_meshes[meter_mesh_number]->local_nodes_end();
-    for (; node_it != end_node_it; ++node_it)
+    // Set the values in the meter meshes.
+    for (unsigned int i = 0; i <= d_num_perim_nodes[meter_idx]; ++i)
     {
-        const Node* node = *node_it;
-        for (unsigned int d = 0; d < NDIM; ++d)
+        Node* node = &d_meter_meshes[meter_idx]->node_ref(i);
+        for (int d = 0; d < NDIM; ++d)
         {
-            const int vel_dof_idx = node->dof_number(velocity_sys_num, d, 0);
-            velocity_solution.set(vel_dof_idx, velocity_coords(vel_dof_idx));
-            const int disp_dof_idx = node->dof_number(displacement_sys_num, d, 0);
-            displacement_solution.set(disp_dof_idx, displacement_coords(disp_dof_idx));
+            (*node)(d) = x_meter_dofs[NDIM * i + d];
+            const int u_dof_idx = d_u_dof_idx[meter_idx][i][d];
+            u_meter_vec->set(u_dof_idx, u_meter_dofs[NDIM * i + d]);
```

Comment:
maybe here you are setting the meter velocity DOF with the mesh velocity DOF?

Reply:
Just pushed a fix.